api/src/opentrons/protocol_engine/state/commands.py

```diff
@@ -416,14 +433,28 @@ def get_next_queued(self) -> Optional[str]:
         if self._state.run_result:
             raise ProtocolEngineStoppedError("Engine was stopped")
 
-        if self._state.queue_status == QueueStatus.INACTIVE:
-            return None
+        # if there is a setup command queued, prioritize it
+        next_setup_cmd = next(iter(self._state.queued_setup_command_ids), None)
+        if next_setup_cmd:
+            return next_setup_cmd
 
-        return next(iter(self._state.queued_command_ids), None)
+        # if the queue is running, return the next protocol command
+        elif self._state.queue_status == QueueStatus.RUNNING:
```

Comment:
not critical! can leave as is! 
```suggestion
        if self._state.queue_status == QueueStatus.RUNNING:
             return next(iter(self._state.queued_command_ids), None)
        # otherwise we've got nothing to do
        return None
```